test/dns-test.js

```diff
@@ -0,0 +1,396 @@
+'use strict';
+
+const assert = require('bsert');
+const {NodeClient, WalletClient} = require('hs-client');
+const StubResolver = require('bns/lib/resolver/stub');
+const bcrypto = require('bcrypto');
+const dnssec = require('bns/lib/dnssec');
+const util = require('bns/lib/util');
+const wire = require('bns/lib/wire');
+const FullNode = require('../lib/node/fullnode');
+const Resource = require('../lib/dns/resource');
+const Network = require('../lib/protocol/network');
+const rules = require('../lib/covenants/rules');
+const common = require('./util/common');
+const {types} = wire;
+
+const json = require('./data/resources-v0.json');
+
+const network = Network.get('regtest');
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  rsNoUnbound: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const nstub = new StubResolver({
+  rd: true,
+  cd: true,
+  edns: true,
+  ednsSize: 4096,
+  maxAttempts: 2,
+  maxTimeout: 3000,
+  dnssec: true,
+  servers: [`127.0.0.1:${network.nsPort}`]
+});
+
+const rstub = new StubResolver({
+  rd: true,
+  cd: true,
+  edns: true,
+  ednsSize: 4096,
+  maxAttempts: 2,
+  maxTimeout: 3000,
+  dnssec: true,
+  servers: [`127.0.0.1:${network.rsPort}`]
+});
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const wallet = wclient.wallet('primary');
+
+const {
+  treeInterval,
+  biddingPeriod,
+  revealPeriod
+} = network.names;
+
+// miner controlled address
+let coinbase;
+
+describe('DNS Servers', function() {
+  this.timeout(15000);
+
+  before(async () => {
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+    await nstub.open();
+    await rstub.open();
+
+    // mine a bunch of blocks
+    const info = await wallet.createAddress('default');
+    coinbase = info.address;
+
+    for (let i = 0; i < 15; i++)
+      await nclient.execute('generatetoaddress', [2, coinbase]);
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+    await nstub.close();
+    await rstub.close();
+  });
+
+  describe('Recursive Resolver', () => {
+    it('should return an A record', async () => {
+      const name = 'google.com.';
+
+      const res = await rstub.lookup(name, types.A);
+
+      const [question] = res.question;
+      assert(typeof question === 'object');
+      assert.equal(question.name, name);
+
+      const [answer] = res.answer;
+      assert(typeof answer === 'object');
+      assert.equal(answer.name, name);
+    });
+  });
+
+  describe('Authoritative Resolver', () => {
+    it('should return no answer for an ICANN rooted name', async () => {
```

Comment:
> I wonder if this test should be duplicated to simulate the scenario in which `com.` actually claims their name on hns.

I like this idea but think that it would expand the scope of this PR